### DISCUSSION TITLE
authorization token validation

### DISCUSSION THREAD

**Discussion by H1Mohamed** [pocketbase/pocketbase#3933] [Q&A]
How to check if a authorization token is valid from the pocketbase sdk logged on as the admin account
I'm running this code in a +server.ts svletekit api file so its fully safe to run

**Comment by ganigeorgiev** [comment]
There is no dedicated endpoint for verifying tokens.

You can create a one-off SDK client and send a `authRefresh()` call (_or `getOne()` if only the authenticated user can view their record_). If it fails - then the token is invalid or expired.

Here is one such example:

```js
const adminClient = new PocketBase(...);

async function checkUserToken(token) {
    // create a one-off user client and load the token in its store
    const tokenClient = new PocketBase(adminClient.baseUrl);
    tokenClient.authStore.save(token);
    
    if (!tokenClient.authStore.isValid) {
        return false; // there is no need for a server-side request as the token is already expired or invalid/empty
    }

    // verify the token
    try {
        await tokenClient.collection("users").authRefresh();
        return true;
    } catch {
        return false;  
    }
}

const isValid = await checkUserToken(token)
```

**Reply by H1Mohamed** [reply]
well i have it so every one can view a user profile.....

**Comment by H1Mohamed** [comment]
Do I make an issue and demand this feature???

**Reply by ganigeorgiev** [reply]
_There were already requests for this and there are no plans for now to add a dedicated token verification endpoint. See for example https://github.com/pocketbase/pocketbase/issues/2997._
